SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/PartitionManager.cpp

```diff
@@ -0,0 +1,557 @@
+	{
+		const ServerWorker& ServerWorkerData = WorkersData.FindChecked(ServerWorkerEntityId);
+		Worker_EntityId SysEntityId = ServerWorkerData.SystemEntityId;
+		const Worker& SystemWorkerData = SystemWorkersData.FindChecked(SysEntityId);
+
+		FLBWorkerHandle ConnectedWorker = MakeShared<FLBWorkerDesc>(FName(SystemWorkerData.WorkerType));
+		ConnectedWorker->State = MakeUnique<FLBWorkerInternalState>();
+		ConnectedWorker->State->ServerWorkerId = ServerWorkerEntityId;
+		ConnectedWorker->State->SystemWorkerId = SysEntityId;
+		ConnectedWorker->State->WorkerType = ConnectedWorker->WorkerType;
+		ConnectedWorker->State->FullWorkerName = SystemWorkerData.WorkerId;
+
+		ConnectedWorkers.Add(ConnectedWorker);
+		ConnectedWorkersThisFrame.Add(ConnectedWorker);
+	}
+
+	void Flush(ISpatialOSWorker& Connection)
+	{
+		if (FirstPartitionId == 0)
+		{
+			return;
+		}
+
+		if (CurPartitionId == FirstPartitionId + k_PartitionsReserveRange)
+		{
+			if (!PartitionReserveRequest.IsSet())
+			{
+				PartitionReserveRequest = Connection.SendReserveEntityIdsRequest(Impl::k_PartitionsReserveRange, RETRY_UNTIL_COMPLETE);
+			}
+			return;
+		}
+
+		for (auto& PartitionEntry : Partitions)
+		{
+			FPartitionInternalState& PartitionState = *PartitionEntry->State;
+			if (!PartitionState.bPartitionCreated)
+			{
+				if (!PartitionState.CreationRequest)
+				{
+					PartitionState.Id = CurPartitionId++;
+					IdToPartitionsMap.Add(PartitionState.Id, PartitionEntry);
+
+					TArray<ComponentData> Components =
+						CreatePartitionEntityComponents(PartitionState.DisplayName, PartitionState.Id, PartitionState.LBConstraint);
+
+					Worker_EntityId PartitionEntity = PartitionState.Id;
+					const Worker_RequestId RequestId =
+						Connection.SendCreateEntityRequest(MoveTemp(Components), PartitionEntity, SpatialGDK::RETRY_UNTIL_COMPLETE);
+					PartitionState.CreationRequest = RequestId;
+					PartitionCreationRequests.Add(RequestId, PartitionEntry);
+				}
+				continue;
+			}
+
+			if (PartitionState.UserAssignment != PartitionState.CurrentAssignment)
+			{
+				if (!PartitionState.AssignmentRequest)
+				{
+					if (ConnectedWorkers.Find(PartitionState.UserAssignment) != nullptr)
+					{
+						Worker_EntityId SystemWorkerEntityId = PartitionState.UserAssignment->State->SystemWorkerId;
+
+						PartitionState.RequestedAssignment = PartitionState.UserAssignment;
+						PartitionState.AssignmentRequest = Connection.SendEntityCommandRequest(
+							SystemWorkerEntityId, CreateClaimPartitionRequest(PartitionState.Id), SpatialGDK::RETRY_UNTIL_COMPLETE, {});
+					}
+				}
+				continue;
+			}
+		}
+	}
+
+	void SetPartitionReady(Worker_EntityId EntityId)
+	{
+		FPartitionHandle* Partition = IdToPartitionsMap.Find(EntityId);
+		if (Partition)
+		{
+			FPartitionInternalState& PartitionState = *(*Partition)->State;
+			PartitionState.bAcked = true;
+		}
+	}
+
+	TArray<ComponentData> CreatePartitionEntityComponents(const FString& PartitionName, const Worker_EntityId EntityId,
```

Comment:
Functionality and naming seems very similar to what's done in the entity factory. Why not consolidate the two?

Reply:
Partitions created in the entity factory and here have different roles, so I find it easier to split the two instead of shoehorning both use cases into one function (given that it also has to branch on the strategy worker flag as well).